src/GraphQL.Tests/Subscription/SubscriptionExecutionStrategyTests.cs

```diff
@@ -0,0 +1,678 @@
+#nullable enable
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+#pragma warning disable IDE0053 // Use expression body for lambda expressions
+
+using GraphQL.Execution;
+using GraphQL.MicrosoftDI;
+using GraphQL.Subscription;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Tests.Subscription;
+
+public class SubscriptionExecutionStrategyTests
```

Comment:
Pretty detailed tests. I did not check them thoroughly, but the general meaning is understandable.